### ISSUE TITLE
Loss value on training and validation set not accurate

### ISSUE THREAD

**Issue by lazarkrstic** [keras-team/keras#13276]
Hi, I notice that value of **loss function** at the end of training model **isn't same** as manually calculated error on same dataset (after model is learned). 

Then I created callback `on_epoch_end` where I measure the error at the and of every epoch. The results are never the same as in training log. Loss function is MAE (mean absolute error). Here is the example:

|Epoch  |loss MAE |  manually MAE |
|--------|------------| ---------------- |
|22|0.11139457036931492| 0.09953024542751052 |
|23|0.09754104633130029|0.08839598977022033|
|24|0.09538907789202652| 0.08944885550643963|

For calculating `MAE manually` I even tried to use `keras.losses.mean_absolute_error` but again 
values are not the same. Manually calculated value are always **lower** than value of loss function.

**Comment by johnlundberg** [comment]
I have also had issues with the val_loss. Eventhough my val_accuracy remains at 0.88, the val_loss, which uses log loss, is extremily low, almost zero. The loss on the training data is 0.28 with 0.88 accuracy, which seems closer to reality.

**Comment by johnlundberg** [comment]
It seems that my issue goes away when downgrading keras to 2.25

**Comment by xiankgx** [comment]
I'm also having some val_loss related issue. In particular, I have a multi-loss model and the weighted validation loss of all the outputs don't add up correctly in val_loss. But loss (training loss) is ok.
